src/codeu/chat/common/Secret.java

```diff
@@ -39,44 +147,6 @@
       compressed[i] = (byte) ((expanded[2 * i] << 4) | expanded[2 * i + 1]);
```

Comment:
It might be more readable to parse and do the shifting to get the compressed version in the same for loop rather than having two.